tools/generator_tools.py

```diff
@@ -509,9 +509,9 @@ def get_dtype_dict (dtype: str, target: str, rank = None, int_len_printf = None)
             'default_prec'                    : '32',
             f'group_{target}_dtype'           : 'int64_t',
             f'group_{target}_h5_dtype'        : 'native_int64',
-            f'group_{target}_f_dtype_default' : 'integer(4)',
-            f'group_{target}_f_dtype_double'  : 'integer(8)',
-            f'group_{target}_f_dtype_single'  : 'integer(4)',
+            f'group_{target}_f_dtype_default' : 'integer(c_int)',
```

Comment:
Same for `c_int` and `c_long` as above.

Reply:
Same here. Old code is correct.

Reply:
@q-posev : Maybe put a comment in the generator to emphasize that we use `integer(8)` and `integer(4)` for good reasons. ;-)